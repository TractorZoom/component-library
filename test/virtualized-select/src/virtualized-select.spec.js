import VirtualizedSelect from '../../../packages/virtualized-select/src';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Virtualized Select', () => {
    it('should render empty', () => {
        // given
        const options = [];

        // when
        const component = renderer.create(
            <VirtualizedSelect label='test' onChange={() => {}} selectedOption={''} options={options} />
        );
        const tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });

    it('should render with 10000 items', () => {
        // given
        const options = [];
        let index = 0;

        while (index < 10000) {
            options.push(`${index}`);
            index++;
        }

        // when
        const component = renderer.create(
            <VirtualizedSelect label='test' onChange={() => {}} selectedOption={options[10]} options={options} />
        );
        const tree = component.toJSON();

        // then
        expect(tree).toMatchSnapshot();
    });
});
