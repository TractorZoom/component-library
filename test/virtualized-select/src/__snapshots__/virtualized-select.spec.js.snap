// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Virtualized Select should render empty 1`] = `
<div
  aria-expanded={false}
  aria-owns={null}
  className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
  label="test"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="combobox"
  selectedOption=""
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="virtualized-select"
      id="virtualized-select-label"
    >
      test
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      onClick={[Function]}
    >
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-controls={null}
        aria-invalid={false}
        autoCapitalize="none"
        autoComplete="off"
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        disabled={false}
        id="virtualized-select"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        required={false}
        spellCheck="false"
        type="text"
        value=""
      />
      <div
        className="MuiAutocomplete-endAdornment"
      >
        <button
          aria-label="Clear"
          className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Clear"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
        <button
          aria-label="Open"
          className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Open"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </span>
        </button>
      </div>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-4"
        >
          <span>
            test
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;

exports[`Virtualized Select should render with 10000 items 1`] = `
<div
  aria-expanded={false}
  aria-owns={null}
  className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
  label="test"
  onClick={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  role="combobox"
  selectedOption="10"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-marginDense MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink={true}
      htmlFor="virtualized-select"
      id="virtualized-select-label"
    >
      test
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      onClick={[Function]}
    >
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-controls={null}
        aria-invalid={false}
        autoCapitalize="none"
        autoComplete="off"
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
        disabled={false}
        id="virtualized-select"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        required={false}
        spellCheck="false"
        type="text"
        value="10"
      />
      <div
        className="MuiAutocomplete-endAdornment"
      >
        <button
          aria-label="Clear"
          className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator MuiAutocomplete-clearIndicatorDirty"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Clear"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Open"
          className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          title="Open"
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-4 PrivateNotchedOutline-legendNotched-5"
        >
          <span>
            test
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
